19266 - Plugged in Company Name Component

ArwenQin opened the pull request:
*Issue #:* /bcgov/entity#19266

*Description of changes:*
- Added the Company Name component
- Validated NR type/state/etc
- Verified support for named or numbered continuation in
- Verified Save and Resume Later

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

ketaki-deodhar commented on src/views/ContinuationIn/ContinuationInBusinessBc.vue:
```diff
@@ -6,7 +6,8 @@
 
-      <div>** Correct Name component goes here **</div>
+      <NameRequestInfo />
+      <NameTranslations />
```
Just porting over Sev's comments on other PR:

Our style is to use the PascalCase version of the component names here, so that VS Code highlights them nicely (and sometimes you can click on them to go to their definition).

Also, don't add divs around them if not needed.

ketaki-deodhar commented:
Sev' comment on other PR

Before your PR is ready for review...

update the app version in the package.json file (you can update the patch version for this PR)
run npm i and also commit the package-lock.json file
run npm run lint and fix any new issues
run npm run test:unit and fix any issues (there should be no previous issues)
in a PR comment, enter /gcbrun, which builds a preview site (URL) with your changes that reviews can run and test with
change this PR to "Ready for review"

bcregistry-sre commented:
Temporary Url for review: https://business-create-dev--pr-688-fyq5ge7j.web.app

severinbeauvais commented on src/views/ContinuationIn/ContinuationInBusinessBc.vue:
```diff
@@ -6,7 +6,8 @@
 
-      <div>** Correct Name component goes here **</div>
+      <NameRequestInfo />
+      <NameTranslations />
```
These need to be in a v-card so that they match the Figma.

Figma:
![image](https://github.com/bcgov/business-create-ui/assets/10002889/0ad5c34e-2f99-4000-9129-8eb5a1e74b79)

Actual:
![image](https://github.com/bcgov/business-create-ui/assets/10002889/bd4a1bfd-dfa2-4fb8-9198-66ff4df22e06)

ArwenQin replied:
Thank you. Updated:
![image](https://github.com/bcgov/business-create-ui/assets/122495122/865a093a-3f00-422d-8c1f-e34a35eadd78)

ArwenQin replied:
Yes, I tested with a named company:
![image](https://github.com/bcgov/business-create-ui/assets/122495122/7ea32f34-3e3b-41ac-abcd-0902abf5e0f4)

ArwenQin replied:
I'm wondering if you feel the 2 spaces should be smaller?
![image](https://github.com/bcgov/business-create-ui/assets/122495122/7d8b4139-032a-4e47-9d31-fc4fbb2266f3)
@severinbeauvais 
Thank you!

severinbeauvais replied:
See how the other filings wrap the components above and you can probably copy it.

severinbeauvais replied:
The v-card looks correct but, yes, I think the top and bottom whitespace should be smaller. Compare with other v-cards.

The named continuation in application looks good 👍

ArwenQin replied:
Thank you Severin. I updated the top and bottom whitespaces:
![image](https://github.com/bcgov/business-create-ui/assets/122495122/d3fc36d8-4d88-4b4a-9090-e8dc0b1d0bf3)

severinbeauvais reviewed (commented):
Have you tested this with a _named_ continuation in application (ie, one that uses a NR to set a company name)?

severinbeauvais commented on src/views/ContinuationIn/ContinuationInBusinessBc.vue:
```diff
@@ -4,9 +4,13 @@
     <section class="mt-10">
       <header id="company-name">
         <h2>Company Name</h2>
+        <p></p>
```
Don't use empty paragraphs for whitespace. Instead, use the Vuetify helper classes to set margins or padding. [link](https://v2.vuetifyjs.com/en/styles/spacing/)

ArwenQin replied:
Got it, thanks. Updated.

severinbeauvais commented on src/views/ContinuationIn/ContinuationInBusinessBc.vue:
```diff
@@ -4,9 +4,13 @@
       </header>
 
-      <div>** Correct Name component goes here **</div>
+      <v-card flat class="py-8 px-6">
```
Does VS Code show you the warnings like the one below?

![image](https://github.com/bcgov/business-create-ui/assets/10002889/e21f677e-fe97-4dd2-9626-32b9639eeef5)

Make sure you have added the following extensions to VS Code:
- ESLint
- Vue - Official

ArwenQin replied:
I just installed the 2 extensions and updated the formatting. Thank you.

bcregistry-sre commented:
Temporary Url for review: https://business-create-dev--pr-688-fyq5ge7j.web.app

severinbeauvais commented:
I think you need to add something to the validation for step 2 (Your Business in B.C.) -- even though I have filled everything in, the review page says that step 2 is unfinished.

Hint: start from the review page to see what it looks at to determine that step 2 is invalid. Then go to step or the store to update whatever is needed so the step is valid when all components on it are valid.

ArwenQin commented:
> Have you tested this with a _named_ continuation in application (ie, one that uses a NR to set a company name)?



> I think you need to add something to the validation for step 2 (Your Business in B.C.) -- even though I have filled everything in, the review page says that step 2 is unfinished.
> 
> Hint: start from the review page to see what it looks at to determine that step 2 is invalid. Then go to step or the store to update whatever is needed so the step is valid when all components on it are valid.

Hi Severin, @severinbeauvais , 
It's an issue with the DefineCompanyStepValidity. 
I updated the ContinuationInBusinessBc.vue file to fix this issue. I added the getter to update the DefineCompanyStepValidity.

severinbeauvais commented on src/views/ContinuationIn/ContinuationInBusinessBc.vue:
```diff
@@ -202,11 +215,12 @@ export default class ContinuationInBusinessBc extends Mixins(CommonMixin) {
+                   (!this.isPremiumAccount || this.getFolioNumberValid)
+    this.setContinuationInBusinessBcValid(allValid)
+    // Update the overall Define Company Step validity
+    this.setDefineCompanyStepValidity(allValid)
```
This looks like the correct solution, but I wonder if we still need `setContinuationInBusinessBcValid()` or whether it can be deleted. The value it sets (getContinuationIn.businessBcValid) is only used for the stepper.

I think you should try to combine this into setDefineCompanyStepValidity().

severinbeauvais replied:
Also, this change broke some unit tests. I would clean this up and then fix the unit tests if they're still broken.

ArwenQin replied:
Hi Severin, 
- I just deleted the setContinuationInBusinessBcValid(), and updated the variables in Stepper (only use isDefineCompanyValid() ).
- I also added the method to validate NameTranslation Component in step 2

severinbeauvais commented on package.json:
Please also commit the package-lock.json file. This is used by the CI process to build the app using exactly the same packages and versions as you locally.

ArwenQin replied:
Checked in package-lock.json

severinbeauvais commented on src/views/ContinuationIn/ContinuationInBusinessBc.vue:
```diff
@@ -142,15 +156,17 @@ export default class ContinuationInBusinessBc extends Mixins(CommonMixin) {
       validBusinessContactForm: this.businessContactFormValid,
-      validFolioReferenceNumber: !this.isPremiumAccount || this.getFolioNumberValid
+      validFolioReferenceNumber: !this.isPremiumAccount || this.getFolioNumberValid,
+      validNameTranslation: this.getNameTranslationsValid
```
Very nice following the existing patterns 👍

severinbeauvais replied:
Oh, darn. This won't work because `name-translation-header` isn't an id on this page.

Also, these flags (and the list on line 156) should be in the order of the components on the page. These are used by `validateAndScroll()` to scroll to the first invalid component.

severinbeauvais replied:
I think it would be OK to scroll to line 5 (change "company-name" to "company-name-header"). And also rename the variable on line 169 accordingly.

ArwenQin replied:
Updated, thank you. Now it works

severinbeauvais commented on src/components/common/Stepper.vue:
```diff
@@ -78,7 +78,6 @@ export default class Stepper extends Vue {
   @Getter(useStore) isAffidavitValid!: boolean
   @Getter(useStore) isAmalgamationInformationValid!: boolean
   @Getter(useStore) isBusySaving!: boolean
-  @Getter(useStore) isContinuationInBusinessBcValid!: boolean
```
Since you've obsoleted this getter (and this was the only place it was used), please remove it from the store.

I think you can also delete `setContinuationInBusinessBcValid()` from the store, and also the respective state variable (see continuation-in-state-interface.ts and state-model.ts).

ArwenQin replied:
Removed the unused getter and variables

severinbeauvais commented:
/gcbrun

severinbeauvais commented:
Looks great so far!

Please finish the unit tests for this page (see ContinuationInBusinessBc.spec.ts).

PS In VS Code, you can search for files by name using Ctrl-P.

bcregistry-sre commented:
Temporary Url for review: https://business-create-dev--pr-688-fyq5ge7j.web.app

severinbeauvais commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-create-dev--pr-688-fyq5ge7j.web.app

severinbeauvais commented:
Code looks good.
Works nicely on preview URLs above.
Just the following:
- [ ] check in package-lock.json
- [ ] update unit tests

ArwenQin commented:
> Code looks good. Works nicely on preview URLs above. Just the following:
> 
> * [ ]  check in package-lock.json
> * [ ]  update unit tests

Thank you so much. I just updated the unit tests and checked in the package-lock.json

severinbeauvais reviewed (approved):
I approve this PR. Arwen, do you have permission to Squash and Merge it?

ArwenQin commented:
> permission to Squash and Merge it

Thanks Severin. But I don't have write access to merge pull yet.
